File: color-shooter/src/core/src/com/salvai/centrum/utils/RandomUtil.java
package com.salvai.centrum.utils;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.math.Vector2;

import java.util.Random;

public class RandomUtil {
    public static Random random = new Random();

    public static Vector2 getRandomEnemyCoordinates() {
        if (random.nextBoolean())
            return new Vector2(random.nextBoolean() ? 0 : Constants.SCREEN_WIDTH, random.nextInt(Constants.SCREEN_HEIGHT));
        else
            return new Vector2(random.nextInt(Constants.SCREEN_WIDTH), random.nextBoolean() ? 0 : Constants.SCREEN_HEIGHT);
    }


    public static int getRandomEnemySpeed(int max) {
        return random.nextInt(max - Constants.ENEMY_MIN_SPEED > 0 ? max - Constants.ENEMY_MIN_SPEED : 1) + Constants.ENEMY_MIN_SPEED;
    }

    public static int getRandomDelay() {
        return random.nextInt(Constants.ENEMY_MAX_DELAY) + Constants.ENEMY_MIN_DELAY;
    }

    public static Color getRandomColor() {
        return GameColorPalette.BASIC[random.nextInt(GameColorPalette.BASIC.length)];
    }

    public static Color getRandomColorNoWhite() {
        return GameColorPalette.BASIC[random.nextInt(GameColorPalette.BASIC.length-1) + 1];
    }

    //BACKGROUND
    public static int getRandomBackgroundStarYCoordinate() {
        return random.nextInt(Constants.SCREEN_HEIGHT);
    }

    public static int getRandomBackgroundStarXCoordinate() {
        return random.nextInt(Constants.SCREEN_WIDTH);
    }

    public static float getRandomAlpha() {
        return random.nextFloat();
    }

    public static int getRandomStarSpeed() {
        return random.nextInt(Constants.MAX_STAR_SPEED - Constants.MIN_STAR_SPEED) + Constants.MIN_STAR_SPEED;
    }
}

File: sn4ke/core/src/com/salvai/snake/utils/RandomUtil.kt
package com.salvai.snake.utils

import com.badlogic.gdx.graphics.Color
import java.util.Random

object RandomUtil {
    var random: Random = Random()

    //BACKGROUND
    @JvmStatic
    fun getRandomBackgroundStarYCoordinate(height: Int): Int {
        return random.nextInt(height)
    }

    @JvmStatic
    val randomCloudSizeFactor: Int
        get() = random.nextInt(3) + 1

    @JvmStatic
    fun getRandomBackgroundStarXCoordinate(width: Int): Int {
        return random.nextInt(width)
    }

    @JvmStatic
    val randomAlpha: Float
        get() = random.nextFloat() * 0.5f

    @JvmStatic
    val randomColor: Color
        get() = Color(Colors.SNAKE_BODY[random.nextInt(Colors.SNAKE_BODY.size)])

    @JvmStatic
    val randomStarSpeed: Int
        get() = random.nextInt(Constants.MAX_BACKGROUND_SPEED - Constants.MIN_BACKGROUND_SPEED) + Constants.MIN_BACKGROUND_SPEED
}
